applications/connectivity_bridge/src/modules/usb_bulk_interface.c

```diff
@@ -173,8 +174,11 @@ static int dap_usb_process(void)
 	static uint8_t tx_buf[USB_BULK_PACKET_SIZE];
 	struct net_buf *buf = k_fifo_get(&dap_rx_queue, K_FOREVER);
 	uint8_t ep = dapusb_config.endpoint[DAP_USB_EP_IN_IDX].ep_addr;
-
+#if defined(CONFIG_BOARD_THINGY91X)
 	len = dap_execute_vendor_cmd(buf->data, tx_buf);
+#else
+	len = dap_execute_cmd(buf->data, tx_buf);
+#endif /* defined(CONFIG_BOARD_THINGY91X) */
```

Comment:
in general, adding board dependencies in code is a bad practice. The clean way to to this is to revert the dependency, ie

- An app level Kconfig (or none)
- Boards that want DAP simply enable DAP, or select app level option
- Application enables DAP based on DAP Kconfigs or the custom one if required.

Reply:
@gmarull is the above response OK?

Reply:
What if this application is used by a customer with its own custom board? Board specific code in applications is never a good idea. Never.

Reply:
This doesn't add board specific code. It reuses code specific to the Thingy:91 X.

Reply:
This is not a sample. If customers want to use it with other boards, they have to port it. Other applications work just the same.

Reply:
Again: board switches at application level are not acceptable. This makes application non-easily portable and makes life unnecessary difficult for end users. The right thing to do this is:

- Add an application level setting, like CONFIG_APP_ENABLE_THIS
- Make sure board overlay enables it if needed

This way, app stays board agnostic, and if any customer porting this app needs the option, they can just touch a conf overlay and not C code.